.github/workflows/ci-prover-e2e.yml

```diff
@@ -51,28 +64,37 @@ jobs:
             --base-token-price-denominator 1 \
             --set-as-default true \
             --ignore-prerequisites \
-            --evm-emulator false
+            --evm-emulator false \
+            --update-submodules=${{matrix.compressor-mode == 'plonk'}}
+
+          ci_run zkstack ecosystem init --dev --update-submodules=${{matrix.compressor-mode == 'plonk'}} --verbose
 
-          ci_run zkstack ecosystem init --dev --verbose
           ci_run zkstack prover init --dev --verbose
-          
           echo "URL=$(grep "http_url" ./chains/proving_chain/configs/general.yaml | awk '{ print $2 }')" >> $GITHUB_ENV
-      - name: Build prover binaries
+
+      - name: Switch verifier to FFLONK interface
+        working-directory: ./chains/proving_chain/configs
+        if: matrix.compressor-mode == 'fflonk'
         run: |
-          ci_run cargo build --release --workspace --manifest-path=prover/Cargo.toml
+          sudo sed -i 's/is_verifier_pre_fflonk: true/is_verifier_pre_fflonk: false/' general.yaml
+          grep "is_verifier_pre_fflonk" general.yaml | awk '{ print $2 }'
+
       - name: Prepare prover subsystem
         run: |
           ci_run zkstack prover init-bellman-cuda --clone --verbose
           ci_run zkstack prover setup-keys --mode=download --region=us --verbose
+
       - name: Run server
         run: |
-          ci_run zkstack server --uring --chain=proving_chain --components=api,tree,eth,state_keeper,commitment_generator,proof_data_handler,vm_runner_protective_reads,vm_runner_bwip &>prover_logs/server.log &
+          ci_run zkstack server --uring --chain=proving_chain --components=api,tree,eth,state_keeper,commitment_generator,proof_data_handler,vm_runner_protective_reads,vm_runner_bwip &>prover_logs_${{matrix.compressor-mode}}/server.log &
       - name: Run Gateway
         run: |
-          ci_run zkstack prover run --component=gateway --docker=false &>prover_logs/gateway.log &
+          ci_run zkstack prover run --component=gateway --docker=false &>prover_logs_${{matrix.compressor-mode}}/gateway.log &
       - name: Run Prover Job Monitor
         run: |
-          ci_run zkstack prover run --component=prover-job-monitor --docker=false &>prover_logs/prover-job-monitor.log &
+          ci_run zkstack prover run --component=prover-job-monitor --docker=false &>prover_logs_${{matrix.compressor-mode}}/prover-job-monitor.log &
+          # Sleeping until the server, gateway and prover job monitor are up
+          ci_run sleep 150
```

Comment:
is there any health check, so we wait without sleep?

Reply:
Made the timeout bigger in the next step, where the health checker is present